src/check/runner/reporter/RunDetails.ts

```diff
@@ -6,7 +6,60 @@ import { ExecutionTree } from './ExecutionTree';
  *
  * A failing property can easily detected by checking the `failed` flag of this structure
  */
-export interface RunDetails<Ts> {
+export type RunDetails<Ts> =
+  | PropertyFailureRunDetails<Ts>

```

Comment:
Be more explicit about the fact it corresponds to a failure in the name of the type